File: SpringBoot/src/main/java/com/project/foret/entity/Comment.java
package com.project.foret.entity;

import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Table(name = "board_comment")
public class Comment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long group_id;
    private String content;
    @Temporal(TemporalType.TIMESTAMP)
    @CreationTimestamp
    private Date reg_date;

    @ManyToOne
    @JoinColumn(name = "board_id")
    private Board board;
    @ManyToOne
    @JoinColumn(name = "writer_id")
    private Member member;
}

File: Android/app/src/main/java/com/project/foret/model/Comment.kt
package com.project.foret.model

data class Comment(
    val id: Long?,
    val group_id: Long?,
    val content: String?,
    val reg_date: String?,
    val member: Member?,
    val board: Board?
) {
    constructor(group_id: Long?, content: String, member: Member, board: Board)
            : this(null, group_id, content, null, member, board)
}
